.github/workflows/design-system.yml

```diff
@@ -0,0 +1,49 @@
+name: Design system deployment
+
+on:
+  push:
+    branches:
+    # TODO: Change to 'main' once tested and released
+      - 'frontend-rework'
+    paths:
+      - 'public/app/frontend/**'
+
+
+jobs:
+  storybook-deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    defaults:
+      run:
+        working-directory: './public/app/frontend'
+    steps:
+      - name: 'Manual checkout'
+        uses: actions/checkout@v4
+
+      - name: 'Set up Node'
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20.x'
+          cache: 'npm'
+          cache-dependency-path: '**/package-lock.json'
+
+      - name: 'Install Node'
+        run: npm install
+
+      - name: 'Build Node'
+        shell: bash
+        run: npm run build-storybook
+
+      - name: "Configuring AWS credentials"
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.S3_ROLE_TO_ASSUME }}
+          aws-region: "eu-west-2"
+
```

Comment:
I opted not to use the storybook-s3 action we discussed as it was just using configure-aws-credentials underneath so this is a bit cleaner.

Reply:
Sounds good!